src/Lift.php

```diff
@@ -60,38 +55,43 @@ public function __construct(
         // Setting up the headers.
         $config['headers']['Content-Type'] = 'application/json';
 
-        // A key consists of your UUID and a MIME base64 encoded shared secret.
-        $this->authKey = new Key($public_key, $secret_key);
+        // Create both unauthenticated and authenticated handler stacks.
+        $handlerStack = isset($config['handler']) ? $config['handler'] : HandlerStack::create();
+        $handlerStack->push($this->addLiftAccountAndSiteId($account_id, $site_id));
 
-        $this->accountId = $account_id;
-        $this->siteId = $site_id;
-
-        // Set our default HandlerStack if nothing is provided
-        if (!isset($config['handler'])) {
-            $config['handler'] = HandlerStack::create();
-        }
-
-        // Add our Account and Site identifiers.
-        $config['handler']->push($this->addLiftAccountAndSiteId());
+        // Both stacks are cloned so they are both completely internal now.
+        $unauthenticatedHandlerStack = clone $handlerStack;
+        $authenticatedHandlerStack = clone $handlerStack;
```

Comment:
I could get away with only cloning once, but I wanted to keep the same behaviour for both handler stacks.